usagov/usagov-2021
# Usagov 1773 left nav module

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1773

## Description

Replaces the core blocks + custom twig templates making the English and Spanish left nav menus with a single module that handles the logic earlier using core Drupal menu APIs.

- fixes an HTML bug where in some cases <a> tags had two class attributes
- fixes another HTML bug where current page in the menu rendered differently in Spanish menus than in English
- Shows all of the parent index page's sibling menu items on states and federal agencies instead of the hard-coded links in a custom template for each


## Type of Changes

- [x ] Frontend (Twig, Sass, JS)
  - Visually there's no change to the menus

## Testing Instructions



### Change Requirements



### Validation Steps

- Manual but could miss something 
    - Open a page, compare a left nav against the same page on another environment or a tome export
    - Test a level 2 page, a page deeper in the hierarchy like in the immigration topic, one or more agency pages, one or more state pages, in both English and Spanish
- Create a new page, add it to a menu, leave it unpublished. Confirm it doesn't show in the menu until you publish the page.

- For thoroughness (if you do this, I can share my html-tidy config)
   - make a tome export of local using the dev branch, rename html/ to html-dev/
   - switch to this branch, make a tome export, rename to html-dev
   - use html-tidy + find command to "normalize" the HTML output in each directory. Use the option to remove indentation. In each HTML folder run a command like
   `find . -name '*.html' -type f -print -exec tidy -config ../tidy-config.txt --warn-proprietary-attributes false -mq '{}' \;`
   - use diff + grep to compare the contents of both directories to confirm the only HTML changes are the fixes listed above. Something like this where the first two `grep -v` commands ignore the cache busting query string drupla adds to CSS and JS files will give you a file to review in a text editor..:
       `diff -brwd --strip-trailing-cr --exclude=_data -I js-view-dom-id --exclude=themes html-old/ html-new/ | grep -v sjpnwi | grep -v sjpja5 | grep -v leftmenusidebarblock | grep -v block-usagov-sidebar-menu | grep -v "nav aria-label='Secondary" | grep -v "nav aria-label=\"Secondary" > nav.diff
`

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
